### DISCUSSION TITLE
will it work for multivariate time series prediction   both regression and classification

### DISCUSSION THREAD

**Discussion by Sandy4321** [bmcclintock/momentuHMM#50] [Q&A] [answered]
great code thanks
may you clarify : 
will it work for multivariate time series prediction   both regression and classification

each row means the time series, and columns represent different continues or/and discrete/category observation.
 
1 
  where all values are continues values 
         weight    height     age   target 
time 1|   56        160    34        1.2
time 2|   77        170    54        3.5
time 3|   87        167     43       0.7
time 4|   55        198    72        0.5
time 5|   88        176     32       2.3
etc

2
or  even will it work for multivariate time series where values are mixture of continues  and categorical values
for example 2 dimensions have continues values and 3 dimensions are categorical values

			color        weight     gender  height  age      target 
time 1|    black       56             m   160    34         yes
time 2|    white       77             f   170    54         no
time 3|   yellow      87             m   167    43          yes
time 4|    white       55             m   198    72         no
time 5|   white       88              f   176    32         yes

etc

**Comment by bmcclintock** [comment, marked as answer]
Yes, as described in the original [article](https://doi.org/10.1111/2041-210X.12995), [vignette](https://cran.r-project.org/web/packages/momentuHMM/vignettes/momentuHMM.pdf), and most of the [examples](https://github.com/bmcclintock/momentuHMM/tree/master/vignettes/examples), `momentuHMM` can handle multiple data streams (e.g. multivariate times series) both for model fitting and for state decoding. The data streams can arise from a broad range of commonly used probability distributions (e.g. beta, categorical, gamma, normal, multivariate normal, Poisson, Weibull); see Tables 2 and 3 in the [vignette](https://cran.r-project.org/web/packages/momentuHMM/vignettes/momentuHMM.pdf) for a complete list of the distributions that are currently supported.
